src/mscorlib/src/System/Runtime/InteropServices/GcHandle.cs

```diff
@@ -197,10 +166,16 @@ public static GCHandle FromIntPtr(IntPtr value)
                     Mda.FireInvalidGCHandleCookieProbe(value);
                     return new GCHandle(IntPtr.Zero);
                 }
+                return new GCHandle(handle);
             }
 #endif
+            return new GCHandle(value);
+        }
 
-            return new GCHandle(handle);
+        internal static object GetTargetFromIntPtr(IntPtr value)
```

Comment:
AssemblyLoadContext does a set of handle double verifies with the target property: `GCHandle.FromIntPtr(...).Target` as seemed to be a particular area where some of the functions where getting marked as no-inline. So I collapsed the two calls and gave its own internal function.

However, I think the most significant change was `m_handle == IntPtr.Zero` to `m_handle.IsNull()` so I could probably revert this and retest?